build/config/substation.libsonnet

```diff
@@ -437,19 +437,33 @@
           kv_store: null,
           close_kv_store: false,
         },
-        get(settings={}): {
-          local type = 'enrich_kv_store_get',
-          local default = $.transform.enrich.kv_store.default { id: $.helpers.id(type, settings) },
+        iget: $.transform.enrich.kv_store.item.get,
+        iset: $.transform.enrich.kv_store.item.set,
+        item: {
+          get(settings={}): {
+            local type = 'enrich_kv_store_get',
+            local default = $.transform.enrich.kv_store.default { id: $.helpers.id(type, settings) },
+
+            type: type,
+            settings: std.prune(std.mergePatch(default, $.helpers.abbv(settings))),
+          },
+          set(settings={}): {
+            local type = 'enrich_kv_store_set',
+            local default = $.transform.enrich.kv_store.default { ttl_key: null, ttl_offset: '0s', id: $.helpers.id(type, settings) },
 
-          type: type,
-          settings: std.prune(std.mergePatch(default, $.helpers.abbv(settings))),
+            type: type,
+            settings: std.prune(std.mergePatch(default, $.helpers.abbv(settings))),
+          },
         },
-        set(settings={}): {
-          local type = 'enrich_kv_store_set',
-          local default = $.transform.enrich.kv_store.default { ttl_key: null, ttl_offset: '0s', id: $.helpers.id(type, settings) },
+        sadd: $.transform.enrich.kv_store.set.add,
+        set: {
```

Comment:
There's two scenarios that I can think of with this:
- You upgrade the entire repo (including this file) and the Jsonnet configurations fail to compile. That's an error that would be raised prior to deployment. It's annoying but it won't impact production deployments. 
- You upgrade the source code (not including this file) and the Jsonnet configurations successfully compile. There's no impact since `enrich_kv_store_set` is still accepted by `transform.New()`.

The configs aren't in scope for SemVer (it's mentioned in VERSIONING) partially for this reason, it's not possible to migrate from (e.g.) `set()` to `set: {}` in Jsonnet while being backwards compatible. An alternative solution is to fork the libsonnet file, but that kicks the problem down the road because this can happen again. I think the only sensible path forward with that is to make the libsonnet file smaller and bring the condition and transform functions into SemVer.

Reply:
> I think the only sensible path forward with that is to make the libsonnet file smaller and bring the condition and transform functions into SemVer.

I agree with this, I forgot they were out of the projects defined SerVer scope but agree that they should be given how it's the projects primary interface across all examples and documentation. Along the SerVer principles, no matter what the breakage is, it should be clearly documented in the changelog and visible with the tagged release. 

Either we should make the breaking change that this is and bump the project to 2.0, or we can use a different container name for the KV operations (such as `transform.enrich.kv_store.set_set(...)`, `transform.enrich.kv_store.set_get(...)`, `transform.enrich.kv_store.item_set(...)`, and `transform.enrich.kv_store.item_get(...)`) to not make this breaking.

Reply:
[This commit](https://github.com/brexhq/substation/pull/215/commits/918de97eb2fa8655ca0119de38d232bba539606f) should make this backcompat with the existing config. We can rename these and make changes to the SemVer docs in a future release.